src/app/api/checkout/route.ts

```diff
@@ -0,0 +1,79 @@
+// NOTE: NextRequest and NextReponse will only work for middleware, and not for 
+// pages or components, so there is a specific command line for it.
+// eslint-disable-next-line
+import { NextRequest, NextResponse } from "next/server";
+import Stripe from 'stripe';
+
+// NOTE: Instead using `require()`, should use `Stripe()`
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY!)
+
+
+// eslint-disable-next-line
+function GET(req: NextRequest, res: NextResponse) {
+    return new Response(JSON.stringify({ code: "Hello Yooshie" }), {
+        status: 200
+    })
+}
+
+async function handler(req: NextRequest, res: NextResponse) {
+    if (req.method === 'GET') {
+        console.log(res)
+        console.log("\n\n\n\n\n\n")
+        console.log("********    here Fack POST")
+
+        return new NextResponse("Hello", {
+            status: 200
+        })
+        // NOTE: Same as the one above.
+        // return new Response(JSON.stringify({ code: "Hello ~ ~ ~ " }), {
+        //     status: 200
+        // })
+    }
+    else if (req.method === 'POST') {
+        try {
+            console.log("********    here success", JSON.stringify(req.headers))
+            // Create Checkout Sessions from body params.
+            const session = await stripe.checkout.sessions.create({
+                line_items: [
+                    {
+                        // Provide the exact Price ID (for example, pr_1234) of the product you want to sell
+                        price: 'price_1Q8xomBmewR921WbEPn531NF',
+                        quantity: 1,
+                    },
+                ],
+                mode: 'payment',
+                // success_url: `${req.headers}/?success=true`,
+                success_url: 'https://example.com/success',
```

Comment:
May want to swap out these place holder URLs with URLs coming from Next - i.e., you want it to send you back to the current page! (Which will be `localhost:3000/gallery/checkout` or something like that)